### PR TITLE
DROOLS-4266 DMN optimize DT ic evaluations

### PR DESCRIPTION
**Pull request by tarilabs**
@etirelli @baldimir @jiripetrlik

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/ast/DMNDTExpressionEvaluator.java [hunk 1]
```diff
@@ -76,18 +77,20 @@ public EvaluatorResult evaluate(DMNRuntimeEventManager dmrem, DMNResult dmnr) {
...
             // need to set the values for in context variables...
-            for ( Map.Entry<String,Object> entry : result.getContext().getAll().entrySet() ) {
+            for (Map.Entry<String, Object> entry : contextValues.entrySet()) {
                 ctx.setValue( entry.getKey(), entry.getValue() );
             }
+            EvaluationContext dtContext = ctx.current();
             for ( int i = 0; i < params.length; i++ ) {
-                EvaluationContextImpl evalCtx = feel.newEvaluationContext(Arrays.asList(events::add), Collections.emptyMap());
-                evalCtx.setValues(result.getContext().getAll());
+                EvaluationContext evalCtx = ctx.current();
```
Comment (etirelli) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/ast/DMNDTExpressionEvaluator.java:88]: where is the ctx.current() defined? is it safe to reuse? I imagine it is, just double checking...

Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/ast/DMNDTExpressionEvaluator.java:88]: It is already safe as-is, but to err on the safe-side even more given is very inexpensive, is even safer as follows this line: https://github.com/kiegroup/drools/pull/2418/files#diff-6db8b96ef6f3c1bdb9e19639abe56e6cR89

**Review by etirelli — APPROVED** [review]

**Comment by etirelli** [comment]
@baldimir @jiripetrlik when you have a chance, can you please review this? it is a small change, but with a significant performance impact.

Thank you!

**Review by baldimir — APPROVED** [review]

**Comment by etirelli** [comment]
@mariofusco can I ask you to merge this, please?

**Pull request merged**